girder-dandi-archive/girder_dandi_archive/schema_validation.py

```diff
@@ -0,0 +1,26 @@
+from semantic_version import Version
+
+from girder.exceptions import ValidationException
+
+from .constants import SCHEMA_MINIMUM_VERSION
+
+
+def validate_schema(event):
+    meta = event.info["meta"]
+    if "dandiset" not in meta:
+        # Do not validate if there is no dandiset metadata property
+        # Only top level dandiset folders should be validated
+        return
+    dandiset = meta["dandiset"]
+    if "schemaVersion" not in dandiset:
+        raise ValidationException(
+            "Missing required metadata field 'schemaVersion'. "
+            "Check that your client is up to date."
+        )
+    schema_version = dandiset["schemaVersion"]
+    if Version(schema_version) < Version(SCHEMA_MINIMUM_VERSION):
```

Comment:
as I have mentioned in the original comment -- `schema_version` could be not a proper semver. Instead of just bailing with a generic exception from semantic_version, IMHO it is better to raise `ValidationException` with a meaningful message